lib/test_scheduler.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright © 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package test_scheduler;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+
+use File::Basename;
+use testapi qw(get_var set_var);
+use main_common 'loadtest';
+use YAML::Tiny;
+
+# use lib '/usr/share/openqa/lib';
+# use OpenQA::Utils qw(asset_type_from_setting locate_asset);
+
+our @EXPORT = qw(load_yaml_schedule);
+
+sub set_variables {
+    my ($var, $value) = @_;
+    $value =~ s/%(.*?)%/get_var($1)/eg;
+    # my $type = asset_type_from_setting($var, $value);
+    # $value = locate_asset($type, $value) if $type;
+    set_var($var, $value);
+}
+
+sub schedule_modules {
+    my ($schedule, $module) = @_;
+    if ($module =~ s/\{\{(.*)\}\}/$1/) {
+        # Module is scheduled conditionally. Need to be parsed. Get condition hash
+        my $condition = $schedule->{conditional_schedule}->{$module};
+        # Iterate over variables in the condition
+        foreach my $var (keys %{$condition}) {
+            next unless my $val = get_var($var);
+            # If value of the variable matched the conditions
+            # Iterate over the list of the modules to be loaded
+            loadtest $_ for (@{$condition->{$var}->{$val}});
+        }
+        next;
```

Comment:
Why do we need this `next` call? I guess `return` would be more intuitive or just switching to `if-else`.